src/browser.js

```diff
@@ -19,25 +19,35 @@ export default function reactExpressMiddlewareGenerator (options = {}) {
 
 	return function reactExpressMiddleware (req, res, next) {
 		res.renderReactComponent = function renderReactComponent (Component, store, done) {
-			// store defaults to res.locals
-			if (typeof store === 'function') {
-				done = store;
-				store = undefined;
-			}
-			if (typeof store !== 'object') {
-				store = res.locals;
+			//
+			// Option 1: Store passed in, Component is a function
+			//
+			if (arguments.length > 1) {
```

Comment:
Yeah, I think so.  Just make sure to write tests for all combinations of the arguments so you cover where it breaks.